### DISCUSSION TITLE
How to use different compression for different pages in multipage tiff

### DISCUSSION THREAD

**Discussion by Ovini** [ImageMagick/ImageMagick#2284] [Help]
I want to create multi page tiff file using a jpeg and a tiff file. The output should be a tiff file with page 1 in jpeg and page 2 in tiff. When I try it with below code page 2 also becomes a jpeg not a tiff.   `  
        
                using (MagickImageCollection images = new MagickImageCollection())
                {
                       MagickImage firstFrame = new MagickImage(@"c:\dips\img1.jpg");
                       firstFrame.Format = MagickFormat.Jpeg;
                       firstFrame.Settings.Compression = CompressionMethod.JPEG;
                       images.Add(firstFrame);

                      MagickImage secondFrame = new MagickImage(@"c:\dips\img2.tif");
                      secondFrame.Format = MagickFormat.Tiff;
                      secondFrame.Settings.Compression = CompressionMethod.Group4;
                      images.Add(secondFrame); 

                      images.Write(@"C:\dips\out.tiff", MagickFormat.Tiff);
               }  
            `
I used https://www.irfanview.com/ to view the page properties of tiff file. Below is the properties of tiff pages. both pages shows the compression as jpeg where I want the 2nd page to be 'CCITT Group 4 Fax Encoding' which is tiff.
![image](https://user-images.githubusercontent.com/9262155/87754816-67da6d00-c823-11ea-9389-45b283cf9f02.png)
Please help.

**Comment by dlemstra** [comment]
I am not sure if you are allowed to set a different compression for each page. I suspect this is an overall setting and the compression that will be used is the one of the first image. Will take a better look at it tonight.

p.s. It seems that you are using Magick.NET you could also ask your question here: https://github.com/dlemstra/Magick.NET/discussions.

**Comment by snibgo** [comment]
From a quick read of coders/tiff.c, I don't think IM can use different compression methods for different images in a tiff file, irrespective of the interface (CLI, MagickCore, etc).

But you can write separate tiff files in IM, with different compression methods, then use "tiffcp" to combine these into a single tiff file, retaining the original compression methods.

For example:
```
magick toes.png -compress zip z.tiff
magick toes.png -compress lzw l.tiff

tiffcp z.tiff l.tiff x.tiff
```
x.tiff then contains two images, with different compression methods.

tiffcp is readily available on Unix systems, or Cygwin for Windows.

**Comment by dlemstra** [comment]
Discussion continued here: https://github.com/dlemstra/Magick.NET/discussions/697.
